hub.py

```diff
@@ -144,26 +149,51 @@ def get_generated_config(self):
             }
         }
 
+    def setup_nfs_share(self):
+        """
+        Create the NFS Share used for user home directories
+
+        Instead of mounting the directory containing *all* hubs' home directories and then using
+        `subDir` to make just the user's directory visible to them, we want to mount the directory
+        containing *just* the home directories of users on this hub. To do so, we must create the
+        directory before the user pods can start - since otherwise there's no NFS share to mount.
+        """
+        # FIXME: This should be provider agnostic
+        zone = self.cluster.spec['gcp']['nfs']['zone']
+        project = self.cluster.spec['gcp']['project']
+        # WARNING: This needs to be idempotent, so watch what you do here.
+        # WARNING: This is a very highly privileged operation, so be careful when you touch this.
+        subprocess.check_call([
+            'gcloud', 'compute', 'ssh', 'nfs-server-01', '--zone', zone, '--project', project, '--',
+            f'sudo mkdir -p {self.nfs_share_name} && sudo chown 1000:1000 {self.nfs_share_name} && sudo ls -ld {self.nfs_share_name}'
+        ])
+
     def deploy(self, auth_provider, proxy_secret_key):
         """
         Deploy this hub
         """
-        client = auth_provider.ensure_client(
-            self.spec['name'],
-            self.spec['domain'],
-            self.spec['auth0']['connection']
-        )
 
         proxy_secret = hmac.new(proxy_secret_key, self.spec['name'].encode(), hashlib.sha256).hexdigest()
+
         secret_values = {
             'jupyterhub':  {
                 'proxy': {
                     'secretToken': proxy_secret
                 },
-                'auth': auth_provider.get_client_creds(client, self.spec['auth0']['connection'])
             }
         }
 
+        # Allow explicilty ignoring auth0 setup
+        if self.spec['auth0'].get('enabled', True):
```

Comment:
So, with this we're ignoring the auth0 setup only for the hubs that have explicitly set `aut0.enabled` False, right?

Reply:
Yep, exactly! It defaults to 'true'